test/Dapr.Client.Test/StateApiTest.cs

```diff
@@ -224,6 +224,135 @@ public async Task SetStateAsync_ThrowsForNonSuccess()
+            options1.Concurrency = ConcurrencyMode.LastWrite;
+            ;
+            var state1 = new StateTransactionRequest("stateKey1", JsonSerializer.SerializeToUtf8Bytes(stateValue1), StateOperationType.Upsert, "testEtag", metadata1, options1);
+
+            int stateValue2 = 100;
+            var state2 = new StateTransactionRequest("stateKey2", JsonSerializer.SerializeToUtf8Bytes(stateValue2), StateOperationType.Delete);
+
+            string stateValue3 = "teststring";
+            var state3 = new StateTransactionRequest("stateKey3", JsonSerializer.SerializeToUtf8Bytes(stateValue3), StateOperationType.Upsert);
+
+            var states = new List<StateTransactionRequest>();
+            states.Add(state1);
+            states.Add(state2);
+            states.Add(state3);
+
+            var task = daprClient.ExecuteStateTransactionAsync("testStore", states);
+
+            // Get Request and validate
+            httpClient.Requests.TryDequeue(out var entry).Should().BeTrue();
+            var transactionRequest = await GrpcUtils.GetRequestFromRequestMessageAsync<Autogenerated.ExecuteStateTransactionRequest>(entry.Request);
+
+            transactionRequest.StoreName.Should().Be("testStore");
+            transactionRequest.Operations.Count.Should().Be(3);
+
+            var req1 = transactionRequest.Operations[0];
+            req1.Request.Key.Should().Be("stateKey1");
+            req1.OperationType.Should().Be(StateOperationType.Upsert.ToString().ToLower());
+            var valueJson1 = req1.Request.Value.ToStringUtf8();
+            var value1 = JsonSerializer.Deserialize<Widget>(valueJson1);
+            value1.Size.Should().Be(stateValue1.Size);
+            value1.Color.Should().Be(stateValue1.Color);
+            req1.Request.Etag.Should().Be("testEtag");
+            req1.Request.Metadata.Count.Should().Be(1);
+            req1.Request.Metadata["a"].Should().Be("b");
+            //optionsFound1.Should().Be(Autogenerated.StateOptions.Concurrency.FirstWrite);
+            req1.Request.Options.Concurrency.Should().Be(2);
+
+            var req2 = transactionRequest.Operations[1];
+            req2.Request.Key.Should().Be("stateKey2");
+            req2.OperationType.Should().Be(StateOperationType.Delete.ToString().ToLower());
+            var valueJson2 = req2.Request.Value.ToStringUtf8();
+            var value2 = JsonSerializer.Deserialize<int>(valueJson2);
+            value2.Should().Be(100);
+
+            var req3 = transactionRequest.Operations[2];
+            req3.Request.Key.Should().Be("stateKey3");
+            req3.OperationType.Should().Be(StateOperationType.Upsert.ToString().ToLower());
+            var valueJson3 = req3.Request.Value.ToStringUtf8();
+            var value3 = JsonSerializer.Deserialize<string>(valueJson3);
+            value3.Should().Be("teststring");
+        }
+
+        [Fact]
+        public async Task ExecuteStateTransactionAsync_CanClearState()
```

Comment:
What makes this test distinct from `ExecuteStateTransactionAsync_CanSaveState`? Both them create a transaction with multiple entries and the verify that the data is serialized correctly. Since you're not hitting an actual data store, the distinction between these two tests is kinda meaningless right?

Reply:
Agree. Will remove